### PR TITLE
MISUV-7394 Dymanic data for poa amount

### PR DESCRIPTION
**Pull request by danielmaltz1**

### CODE REVIEW

File: app/connectors/FinancialDetailsConnector.scala [hunk 1]
```diff
@@ -91,15 +92,17 @@ class FinancialDetailsConnector @Inject()(val http: HttpClient,
...
+                         (implicit headerCarrier: HeaderCarrier, mtdItUser: MtdItUser[_]): Future[FinancialDetailsResponseModel] = {
 
     val dateFrom: String = (taxYear - 1).toString + "-04-06"
     val dateTo: String = taxYear.toString + "-04-05"
 
     val url = getChargesUrl(nino, dateFrom, dateTo)
     Logger("application").debug(s"GET $url")
 
-    http.GET[HttpResponse](url)(httpReads, headerCarrier, implicitly) map { response =>
+    val hc = checkAndAddTestHeader(mtdItUser.path, headerCarrier, appConfig.poaAdjustmentOverrides(), "afterPoaAmountAdjusted")
```
Comment (avelx) [inline app/connectors/FinancialDetailsConnector.scala:103]: * we might need to raise a tech debt to move all this kind of transformation to a different level, like => HeaderExtractor etc;

Comment (danielmaltz1) [reply app/connectors/FinancialDetailsConnector.scala:103]: I think we only want this to be called in these 2 specific scenarios (at least, at the moment), so I'm not sure if moving it to the HeaderExtractor is the best place

Comment (avelx) [reply app/connectors/FinancialDetailsConnector.scala:103]: I am going to create a tech debt story for you as I think connectors should not include this type of logic

**Review by avelx — APPROVED** [review]
approved with comments

**Pull request merged**